src/main/common/draft/draftStoreClient.ts

```diff
@@ -0,0 +1,75 @@
+import { Draft } from 'app/models/draft'
+import { DraftDocument } from 'app/models/draftDocument'
+import { CoreOptions, RequestAPI } from 'request'
+import { RequestPromise } from 'request-promise-native'
+import moment = require('moment')
+
+export default class DraftStoreClient<T extends DraftDocument> {
+  constructor (public endpointURL: string,
+               public serviceAuthToken: string,
+               public request: RequestAPI<RequestPromise, CoreOptions, CoreOptions>) {
+    this.endpointURL = endpointURL
+    this.serviceAuthToken = serviceAuthToken
+    this.request = request
+  }
+
+  find (query: { [key: string]: string }, userAuthToken: string, deserializationFn: (value: any) => T): Promise<Draft<T>[]> {
+    const { type, ...qs } = query
+    const endpointURL: string = `${this.endpointURL}/drafts`
+
+    return this.request
+      .get(endpointURL, {
+        qs: qs,
+        headers: this.authHeaders(userAuthToken)
+      })
+      .then((response: any) => {
+        return response.data
+          .filter(draft => type ? draft.type === type : true)
+          .map(draft => {
+            return new Draft(
+              draft.id,
+              draft.type,
+              deserializationFn(draft.document),
+              moment(draft.created),
+              moment(draft.updated)
+            )
+          })
+      })
+  }
+
+  save (draft: Draft<T>, userAuthToken: string): Promise<void> {
```

Comment:
api returns ID of newly created draft in headers, it would be good to return it in the client as well.

Reply:
in the headers? why doesn't it just return it directly?

Reply:
https://hmcts.github.io/restful-api-standards/#success-codes